base_bank_from_iban/models/res_partner_bank.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2017 Tecnativa - Carlos Dauden <carlos.dauden@tecnativa.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl-3).
+
+from odoo import api, models
+
+from odoo.addons.base_iban.models.res_partner_bank import (
+    _map_iban_template,
+    normalize_iban,
+    pretty_iban,
+)
+
+
+class ResPartnerBank(models.Model):
+    _inherit = "res.partner.bank"
+
+    @api.onchange("acc_number", "acc_type")
+    def _onchange_acc_number_base_bank_from_iban(self):
```

Comment:
Instead of the deprecated onchange, please use computed writable field on `bank_id` with the following convention:

https://github.com/OCA/odoo-community.org/pull/50/files

Reply:
And please check if with code creation, the field is computed as well.